lib/nrf_modem_lib/nrf91_sockets.c

```diff
@@ -458,9 +440,12 @@ static int nrf91_socket_offload_setsockopt(void *obj, int level, int optname,
 	}
 
 	if ((level == SOL_SOCKET) && ((optname == SO_RCVTIMEO) ||
-		(optname == SO_SNDTIMEO))) {
-		nrf_timeo.tv_sec = ((struct timeval *)optval)->tv_sec;
-		nrf_timeo.tv_usec = ((struct timeval *)optval)->tv_usec;
+				      (optname == SO_SNDTIMEO))) {
+		if (optval != NULL) {
+			nrf_timeo.tv_sec = ((struct timeval *)optval)->tv_sec;
+			nrf_timeo.tv_usec = ((struct timeval *)optval)->tv_usec;
+		}
```

Comment:
Thank you for pointing this out. Added initialization to the `nrf_timeo` struct for good measure and moved the code that updates `nrf_optval` and `nrf_optlen` inside the `if` block so that we don't override the parameters when `optval` is NULL.